.github/workflows/test.yaml

```diff
@@ -1,77 +1,78 @@
-name: "test"
+name: "scaffolding-ci"
 
 on:
   # Trigger the workflow on push or pull request,
   # but only for the main branch
   push:
-    branches: [ develop, develop-0.1, develop-0.2 ]
+    branches: [develop, develop-0.1, develop-0.2]
   pull_request:
-    branches: [ develop, develop-0.1, develop-0.2 ]
+    branches: [develop, develop-0.1, develop-0.2]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
 
 jobs:
-  test_fmt_lint:
+  cargo_test:
+    name: cargo_test
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-
-      - name: Maximize build space
-        if: runner.os == 'Linux'
-        uses: AdityaGarg8/remove-unwanted-software@v2
+      - uses: dtolnay/rust-toolchain@stable
         with:
-          remove-dotnet: 'true'
-          remove-android: 'true'
-          remove-codeql: 'true'
-          remove-docker-images: 'true'
+          toolchain: 1.80.0
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo test --no-fail-fast
 
-      - name: Install nix
-        uses: cachix/install-nix-action@v27
+  rustfmt:
+    name: rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
         with:
-          install_url: https://releases.nixos.org/nix/nix-2.24.4/install
+          toolchain: 1.80.0
+          components: rustfmt
+      - run: cargo fmt --all -- --check
 
-      - uses: cachix/cachix-action@v15
+  clippy:
+    name: clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
         with:
-          name: holochain-ci
-
-      - name: rustfmt check
-        run: |
-          cd $GITHUB_WORKSPACE
-          nix develop .#ci --command cargo fmt --all --check
-
-      - name: clippy check
-        run: |
-          cd $GITHUB_WORKSPACE
-          nix develop .#ci --command cargo clippy --all-features --all-targets -- -D warnings
-
-      - name: unittests
-        run: |
-          cd $GITHUB_WORKSPACE
-          nix develop .#ci --command cargo test --no-fail-fast
+          toolchain: 1.80.0
+          components: clippy
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo clippy -- -D warnings
 
   testbuild:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        template: [ svelte, vue, lit, react, vanilla ]
+        template: [svelte, vue, lit, react, vanilla]
```

Comment:
These will still run in parallel and rebuild the scaffolding a bunch of times, so the recommendation to build it once for Nix isn't done yet, is that for a follow up PR?

Reply:
Hmmm hadn't considered this. I think I can do this separately though

Reply:
As in, a problem to solve on another PR?

Reply:
Yes, I will need to weigh the impact of this. Probably will need to run the jobs sequentially so that only the first job builds scaffolding once. Running a job where there was a cache miss usually will take 35 minutes, then subsequent jobs using the cache from what I see take <= 10 minutes. So overall the jobs will succeed after (35min + <=10min * n) where the n is the number of jobs, this is much longer that running them in parallel which would take 35 minutes, much faster but not optimal. Anyway, yes, I will do this with a follow up PR, will keep the issue open.

Reply:
Just pushed this commit though, I can keep it as part of this PR if this actually solves the problem

https://github.com/holochain/scaffolding/pull/414/commits/6853399e52585320b4e342cd6951d61004dc109c